test/services/commits_runner_test.rb

```diff
@@ -0,0 +1,45 @@
+require 'test_helper'
+
+class CommitsRunnerTest < ActiveSupport::TestCase
+  include ActiveJob::TestHelper
+
+  setup do
+    @repo = create(:repo, name: 'rails')
+  end
+
+  test '#run' do
+    commits = [
+      {
+        sha: '12345',
+        message: 'My beautiful commit message',
+        repo: {
+          id: @repo.id,
+          name: @repo.name
+        },
+        author: {
+          name: 'bmarkons'
+        },
+        url: 'https://github.com/commit',
+        created_at: 12345
+      }
+    ]
+
+    CommitsRunner.run(commits)
+
+    expect_created(commits)
+    assert_enqueued_jobs(commits.count)
+  end
+
+  def expect_created(commits_hashes)
+    commits_hashes.each do |hash|
+      commit = Commit.find_by!(sha1: hash[:sha])
+
+      assert_equal commit.sha1, hash[:sha]
```

Comment:
I guess you should put arguments the other way around to have a non-misleading error message in case things are not equal.

~~~ruby
assert_equal hash[:sha], commit.sha1
~~~